CMakeLists.txt

```
@@ -100,7 +100,7 @@ elseif(ANDROID)
     message(FATAL_ERROR "Android build not supported!")
 elseif(APPLE)
     target_compile_definitions(platform_wsi INTERFACE VK_USE_PLATFORM_MACOS_MVK VK_USE_PLATFORM_METAL_EXT)
-elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|BSD")
+elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|BSD|GNU|DragonFly")
```

Question @jbeich in CMake 3.25 they introduce LINUX and BSD

https://cmake.org/cmake/help/latest/variable/LINUX.html
https://cmake.org/cmake/help/latest/variable/BSD.html

Would the following check be sufficient in the future?
```
elseif(LINUX OR BSD)
```

---

Maybe stick `BSD` and `DragonFly` closer together due to being related like
```suggestion
elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|DragonFly|BSD|GNU")
```

`elseif(LINUX OR BSD)` works fine at least on FreeBSD even without `cmake_minimum_required(VERSION 3.25)`. However, GNU/Hurd may still require `CMAKE_SYSTEM_NAME` check.

---

Updated to `Linux|BSD|DragonFly|GNU`